src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String MASSAGE_FOR_REPORT = "Report for period ";
+    private static final String DASH = " - ";
+    private static final String SPACE = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int periodSalary = 0;
+        StringBuilder builder = new StringBuilder();
+        builder.append(MASSAGE_FOR_REPORT).append(dateFrom)
+                .append(DASH).append(dateTo).append(System.lineSeparator());
+        for (int namesIndex = 0; namesIndex < names.length; namesIndex++) {
+            for (int dataIndex = 0; dataIndex < data.length; dataIndex++) {
+                if (names[namesIndex].equals(getNameFromData(data[dataIndex]))) {
+                    String dataArray = data[dataIndex].substring(0, data[dataIndex].indexOf(SPACE));
+                    if (compareDates(dateFrom, dataArray, dateTo)) {
+                        periodSalary += getSalary(data[dataIndex]);
+                    }
+                }
+            }
+            if (namesIndex < names.length - 1) {
+                builder.append(names[namesIndex]).append(DASH)
+                        .append(periodSalary).append(System.lineSeparator());
+            } else {
+                builder.append(names[namesIndex]).append(DASH).append(periodSalary);
+            }
+            periodSalary = 0;
+        }
+        return builder.toString();
+    }
+
+    public boolean compareDates(String dateFrom, String dateFromArray, String dateTo) {
+        LocalDate dateOne = LocalDate.from(formatter.parse(dateFrom));
+        LocalDate dateTwo = LocalDate.from(formatter.parse(dateFromArray));
+        LocalDate dateThree = LocalDate.from(formatter.parse(dateTo));
```

Comment:
You are parsing dateFrom and dateTo each time the method is called. I'd recommend parsing them in advance before loop